### PR TITLE
Dan/manual svc delete cli

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description: What code changed, and why?

Adds the necessary API to support the manual service deletion flow described in #1192 and provides a CLI to use it.

### :chains: Related Resources

#1192
#2488 

### :athletic_shoe: How to Build and Test the Change

* *Build:* automate-gateway, automate-cli, applications-service
* Start all services and run `applications_populate_database`
* `chef-automate applications show-svcs` shows all the services
* `chef-automate applications show-svcs -a "*" -b "*" -c "*" -e "*" -g "*" -o "*" -n "*" -s "*" -v "*" -D` uses all the filtering options (this will probably not return any services unless you waited for them to become disconnected)
* Experiment with any of the filter options (see what they mean with `--help`)
* Delete services with `chef-automate applications remove-svcs`, they should not show up in the `show-svcs` command after.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable

```
chef-automate applications show-svcs -a "*" -b "*" -c "*" -e "*" -g "*" -o "*" -n "*" -s "*" -v "*" | head -5
id   	svc.group    	release                          	FQDN                	app:env  	status  
17741	redis.default	custom/redis/1.0.1/20190115189830	ebnru2er.example.net	demo:demo	CRITICAL
17695	nginx.default	custom/nginx/2.0.0/20190115189976	7810o1k8.example.net	demo:demo	CRITICAL
17734	redis.default	custom/redis/1.0.1/20190115189830	djfjxnrq.example.net	demo:demo	CRITICAL
17676	nginx.default	custom/nginx/2.0.0/20190115189976	23hacbg2.example.net	demo:demo	CRITICAL
17670	billing-svc.default	custom/billing-svc/1.3.4/20190115182382	jb0dbg8y.example.net	billing-backend:qa	CRITICAL
```

```
[258][default:/src:0]# chef-automate applications remove-svcs -a "m*" -b "*" -c "*" -e "p*" -g "*" -o "*" -n "mobile-api-auth*" -s "*" -v "*" 
id   	svc.group              	release                                    	FQDN                	app:env                  	status
18542	mobile-api-auth.default	custom/mobile-api-auth/2.0.8/20190115181192	bllhcayc.example.net	mobile-app-api:production	OK

The above 1 services will be deleted. Do you wish to continue? (y/n)
y
Removed 1 services
[259][default:/src:0]# chef-automate applications show-svcs -a "m*" -b "*" -c "*" -e "p*" -g "*" -o "*" -n "mobile-api-auth*" -s "*" -v "*" 
id	svc.group	release	FQDN	app:env	status
```

### CODE REVIEW

**Comment by vjeffrey** [comment]
i really appreciate the detailed repro steps, **thank you** for always including those!!
i'll go through and review locally after demos

File: components/applications-service/pkg/server/server.go [hunk 1]
```diff
@@ -335,6 +335,29 @@ func (app *ApplicationsServer) DeleteDisconnectedServices(ctx context.Context,
...
 }
 
+func (app *ApplicationsServer) DeleteServicesByID(ctx context.Context,
+	request *applications.DeleteServicesByIDReq) (*applications.ServicesRes, error) {
+	svcIDs := request.GetIds()
+	if svcIDs == nil {
+		return new(applications.ServicesRes),
+			status.Error(codes.InvalidArgument, "Must specify service IDs to delete")
+	}
+	if len(svcIDs) == 0 {
```
Comment (vjeffrey) [inline components/applications-service/pkg/server/server.go]: i'm curious why the nil check == return error but the 0 id check returns empty?

File: .studio/applications-service [hunk 2]
```diff
@@ -240,6 +240,12 @@ function applications_publish_supervisor_message() {
   applications-publisher --internal-nats --port 10140 "$@"
 }
 
+function applications_random_fqdn() {
```
Comment (afiune) [inline .studio/applications-service]: tenor-130446749

File: components/applications-service/pkg/server/server.go (hunk 1 above)
Comment (afiune) [reply components/applications-service/pkg/server/server.go]: I had the same question, I remember defaulting to return `new()` for consistency.

File: components/applications-service/pkg/storage/postgres/services.go [hunk 3]
```diff
@@ -164,6 +165,34 @@ RETURNING s.id
 	, s.health_check_exit_status as health_check_exit_status
 `
 
+	deleteServicesByID = `
+   DELETE FROM service_full AS s
```
Comment (afiune) [inline components/applications-service/pkg/storage/postgres/services.go]: (nit): extra tab.
```suggestion
DELETE FROM service_full AS s
```

File: components/applications-service/pkg/storage/postgres/services.go [hunk 4]
```diff
@@ -164,6 +165,34 @@ RETURNING s.id
...
+    WHERE s.id IN (%s)
```
Comment (afiune) [inline components/applications-service/pkg/storage/postgres/services.go]: (nit): extra tab.
```suggestion
WHERE s.id IN (%s)
```

File: components/applications-service/pkg/server/server.go (hunk 1 above)
Comment (danielsdeleo) [reply components/applications-service/pkg/server/server.go]: @afiune I think the question is why I chose to return `status.Error(codes.InvalidArgument, ....` when `request.GetIds()` is nil, but return an empty response for an empty request. The answer to that is that the nil case feels more like a malformed/incompletely formed request, whereas an empty request is kinda silly but correctly formed. You could also imagine that a user writes some periodic cleanup script and doesn't think to skip the delete request when there's zero services to cleanup; failing the request might trigger some alerts or something for little reason.

I'll change the ` &applications.ServicesRes{}` to `new` so it matches.

Comment (vjeffrey) [reply components/applications-service/pkg/server/server.go]: that makes sense, thanks for explaining!

Comment (vjeffrey) [reply components/applications-service/pkg/server/server.go]: that makes sense, thanks for explaining!

**Pull request merged**